daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/value/Value.scala

```diff
@@ -76,6 +76,18 @@ sealed abstract class Value[+Cid] extends CidContainer[Value[Cid]] with Product
               go(exceededNesting, errs, values.toImmArray.map(v => (v, newNesting)) ++: vs)
             }
 
+          case ValueBuiltinException(_, value) =>
```

Comment:
side note: the logic for checking against max nesting seems to be duplicated quite a bit. maybe worth factoring it out at some point.